audio_service/lib/audio_service.dart

```diff
@@ -295,42 +311,42 @@ class Rating {
   final RatingStyle _type;
   final dynamic _value;
 
-  const Rating._internal(this._type, this._value);
+  const Rating._(this._type, this._value);
 
   /// Creates a new heart rating.
   const Rating.newHeartRating(bool hasHeart)
-      : this._internal(RatingStyle.heart, hasHeart);
+      : this._(RatingStyle.heart, hasHeart);
 
   /// Creates a new percentage rating.
-  factory Rating.newPercentageRating(double percent) {
-    if (percent < 0 || percent > 100) throw ArgumentError();
-    return Rating._internal(RatingStyle.percentage, percent);
-  }
+  const Rating.newPercentageRating(double percent)
+      : assert(percent >= 0 && percent <= 100,
+            'Percentage must be in range from 0 to 100'),
+        _type = RatingStyle.percentage,
+        _value = percent;
 
   /// Creates a new star rating.
-  factory Rating.newStarRating(RatingStyle starRatingStyle, int starRating) {
-    if (starRatingStyle != RatingStyle.range3stars &&
-        starRatingStyle != RatingStyle.range4stars &&
-        starRatingStyle != RatingStyle.range5stars) {
-      throw ArgumentError();
-    }
-    if (starRating > starRatingStyle.index || starRating < 0)
-      throw ArgumentError();
-    return Rating._internal(starRatingStyle, starRating);
-  }
+  Rating.newStarRating(RatingStyle style, int rating)
+      : assert(
+            style == RatingStyle.range3stars ||
+                style == RatingStyle.range4stars ||
+                style == RatingStyle.range5stars,
```

Comment:
I wouldn't personally put the comma here. Although I don't have a strong opinion either way, I value small diffs more ;-) Unless the change has clear and demonstrable value.